### PR TITLE
darglint2: Cleanup configuration code.

### PR DESCRIPTION
**Pull request by real-yfprojects**
* Cleanup `message_template` configuration option.
* Add code comment to `default_config` fixture.
* Cleanup `Configuration` class.
* Fix tests broken by missing `message_template` parameter.

### CODE REVIEW

File: darglint2/config.py [hunk 1]
```diff
@@ -78,23 +79,46 @@ def from_string(cls, level: str) -> "LogLevel":
...
+        message_template: the template with which to format the errors.
+        style: The style of docstring.
+        strictness: The minimum strictness to allow.
+        ignore_regex: A regular expression which enables ignoring
+            functions/methods by name.
+        ignore_raise: A list of exceptions that don't need to be
+            documented.
+        ignore_properties: Bool indicating whether to ignore properties functions
+            or not.
+        enable: A list of of error codes that are disabled by default.
```
Comment (akaihola) [inline darglint2/config.py:97]: Is this accurate? Should it say `...error codes to enable that are disabled by default`? Or perhaps `...are enabled by default`?

File: darglint2/config.py [hunk 2]
```diff
@@ -78,23 +79,46 @@ def from_string(cls, level: str) -> "LogLevel":
...
 
 
-class Configuration(object):
+class Configuration:
+    """
+    A dataclass representing a configuration.
+
+    Attributes:
+        ignore: A list of error codes to ignore.
+        message_template: the template with which to format the errors.
```
Comment (akaihola) [inline darglint2/config.py:88]:
```suggestion
        message_template: The template with which to format the errors.
```

File: darglint2/config.py [hunk 3]
```diff
@@ -78,23 +79,46 @@ def from_string(cls, level: str) -> "LogLevel":
...
+    """
+
     def __init__(
         self,
-        ignore: List[str],
-        message_template: Optional[str],
-        style: DocstringStyle,
-        strictness: Strictness,
+        *,
+        ignore: List[str] = [],
```
Comment (akaihola) [inline darglint2/config.py:108]: Should not use mutable default value. Better to use `Optional[List[str]] = None` and `self.ignore = ignore or []`.

Same for `ignore_raise` and `enable` below.

File: darglint2/config.py [hunk 4]
```diff
@@ -137,6 +159,21 @@ def log_level(self, log_level: LogLevel) -> None:
...
+        For use in constructing a cached `errors_to_ignore` value.
+        Since this value could be used frequently, it makes
+        sense to cache this value.
+
+        Returns:
+            The errors to ignore, including the default errors.
+
+        """
+        disabled = DEFAULT_DISABLED - set(self.enable)
+        return self.ignore + list(disabled)
```
Comment (akaihola) [inline darglint2/config.py:175]: This list may have duplicates, right?

Also, for repeatability and unit testing purposes, `sorted(disabled)` might be better?

Or change the return value to a set if order doesn't matter?

File: darglint2/config.py [hunk 5]
```diff
@@ -137,6 +159,21 @@ def log_level(self, log_level: LogLevel) -> None:
         logger = get_logger()
         logger.setLevel(log_level.value)
 
+    @property
+    def errors_to_ignore(self) -> List[str]:
+        """Update the errors to ignore, accounding for defaults.
+
+        For use in constructing a cached `errors_to_ignore` value.
+        Since this value could be used frequently, it makes
+        sense to cache this value.
```
[the remark this answers is not in the document]
Comment (real-yfprojects) [reply darglint2/config.py:168]: Which option do you prefer?

[the remark this answers is not in the document]
Comment (akaihola) [reply darglint2/config.py:168]: Could I leave this choice up to you? My day job commitments currently drain my extra energy... Thanks!

[the remark this answers is not in the document]
Comment (real-yfprojects) [reply darglint2/config.py:168]: I chose to cache it in `IntegrityChecker`.

**Comment by real-yfprojects** [comment]
If you have the time and energy I would appreciate a review from you @akaihola. Otherwise merging this without a review doesn't present a problem either.

File: darglint2/config.py [hunk 6]
```diff
@@ -78,27 +79,50 @@ def from_string(cls, level: str) -> "LogLevel":
             raise ValueError('Unrecognized log level, "{}"'.format(level))
 
 
-class Configuration(object):
+class Configuration:
+    """
+    A dataclass representing a configuration.
```
Comment (akaihola) [inline darglint2/config.py:84]: Is it confusing to call this a "dataclass" when a concept with that name exists in the Python standard library, but this class doesn't actually use it?

Comment (real-yfprojects) [reply darglint2/config.py:84]: Good question

Comment (real-yfprojects) [reply darglint2/config.py:84]: It kind of is. Do you have an idea for a different phrasing?

**Review by akaihola — APPROVED** [review]
All looks good!

**Pull request merged**